crates/cext/src/transpiler/passes/vf2.rs

```diff
@@ -0,0 +1,230 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use hashbrown::HashMap;
+
+use crate::pointers::const_ptr_as_ref;
+
+use qiskit_circuit::circuit_data::CircuitData;
+use qiskit_circuit::dag_circuit::DAGCircuit;
+use qiskit_circuit::{PhysicalQubit, VirtualQubit};
+use qiskit_transpiler::passes::vf2_layout_pass;
+use qiskit_transpiler::target::Target;
+
+/// The result from ``qk_transpiler_pass_standalone_vf2_layout()``.
+pub struct VF2LayoutResult(Option<HashMap<VirtualQubit, PhysicalQubit>>);
+
+/// @ingroup QkVF2LayoutResult
+/// Check whether a result was found.
+///
+/// @param layout a pointer to the layout
+///
+/// @returns ``true`` if the ``qk_transpiler_pass_standalone_vf2_layout()`` run found a layout
+///
+/// # Safety
+///
+/// Behavior is undefined if ``layout`` is not a valid, non-null pointer to a
+/// ``QkVF2LayoutResult``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_vf2_layout_has_match(layout: *const VF2LayoutResult) -> bool {
+    let layout = unsafe { const_ptr_as_ref(layout) };
+    layout.0.is_some()
+}
+
+/// @ingroup QkVF2LayoutResult
+/// Get the number of virtual qubits in the layout.
+///
+/// @param layout a pointer to the layout
+///
+/// @returns The number of virtual qubits in the layout
+///
+/// # Safety
+///
+/// Behavior is undefined if ``layout`` is not a valid, non-null pointer to a
+/// ``QkVF2LayoutResult``. The result must have a layout found.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_vf2_layout_num_qubits(layout: *const VF2LayoutResult) -> u32 {
+    let layout = unsafe { const_ptr_as_ref(layout) };
+    let Some(ref layout) = layout.0 else {
+        panic!("There was no layout found");
```

Comment:
Just a thought, but I would expect the language we use when panicking to be a little less like a user error message and more "you have a bug" vibes. E.g.

```suggestion
        panic!("Invalid call for empty layout result");
```